libs/langchain-core/src/messages/base.ts

```diff
@@ -378,6 +378,41 @@ export abstract class BaseMessage<
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     return `${(this.constructor as any).lc_name()} ${printable}`;
   }
+
+  /**
+   * Protected method for derived classes to add type-specific details
+   * to the pretty print output. Returns an array of strings to be inserted
+   * between the title and content sections.
+   */
+  protected _prettyPrintDetails(): string[] {
+    return [];
+  }
+
+  prettyPrint(): string {
```

Comment:
note that in python it directly prints instead of returning a string. Should we match that? If so I'd be inclined to have that be a simple wrapper of this func for testability